src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,398 @@
+    if (componentState.serverSaveTime != null) {
+      const formattedDateTime = this.utilService.convertMillisecondsToFormattedDateTime(
+        componentState.serverSaveTime
+      );
+      row[columnNameToNumber.get('Server Timestamp')] = formattedDateTime;
+    }
+    if (componentState.clientSaveTime != null) {
+      const clientSaveTime = new Date(componentState.clientSaveTime);
+      if (clientSaveTime != null) {
+        const clientSaveTimeString =
+          clientSaveTime.toDateString() + ' ' + clientSaveTime.toLocaleTimeString();
+        row[columnNameToNumber.get('Client Timestamp')] = clientSaveTimeString;
+      }
+    }
+    row[columnNameToNumber.get('Node ID')] = componentState.nodeId;
+    row[columnNameToNumber.get('Component ID')] = componentState.componentId;
+    row[columnNameToNumber.get('Step Title')] = this.projectService.getNodePositionAndTitle(
+      componentState.nodeId
+    );
+    const componentPartNumber =
+      this.projectService.getComponentPosition(componentState.nodeId, componentState.componentId) +
+      1;
+    row[columnNameToNumber.get('Component Part Number')] = componentPartNumber;
+    row[columnNameToNumber.get('Component Type')] = this.component.type;
+    if (this.component.prompt != null) {
+      let prompt = this.utilService.removeHTMLTags(this.component.prompt);
+      prompt = prompt.replace(/"/g, '""');
+      row[columnNameToNumber.get('Component Prompt')] = prompt;
+    }
+    const studentData = componentState.studentData;
+    if (studentData != null) {
+      row[columnNameToNumber.get('Student Data')] = studentData;
+      const isCorrect = studentData.isCorrect;
+      if (isCorrect != null) {
+        if (isCorrect) {
+          row[columnNameToNumber.get('Is Correct')] = 1;
+        } else {
+          row[columnNameToNumber.get('Is Correct')] = 0;
+        }
+      }
+    }
+    const revisionCounter = this.getRevisionCounter(
+      componentRevisionCounter,
+      componentState.nodeId,
+      componentState.componentId
+    );
+    if (componentState.revisionCounter == null) {
+      /*
+       * use the revision counter obtained from the componentRevisionCounter
+       * mapping. this case will happen when we are exporting all student
+       * work.
+       */
+      row[columnNameToNumber.get('Component Revision Counter')] = revisionCounter;
+    } else {
+      /*
+       * use the revision counter from the value in the component state.
+       * this case will happen when we are exporting latest student work
+       * because the revision counter needs to be previously calculated
+       * and then set into the component state
+       */
+      row[columnNameToNumber.get('Component Revision Counter')] = componentState.revisionCounter;
+    }
+    this.incrementRevisionCounter(
+      componentRevisionCounter,
+      componentState.nodeId,
+      componentState.componentId
+    );
+    const isSubmit = componentState.isSubmit;
+    if (isSubmit) {
```

Comment:
No need to create isSubmit variable here, since it's only being used once?
```suggestion
    if (componentState.isSubmit) {
```